grails-app/controllers/org/broadinstitute/orsp/api/SampleConsentLinkController.groovy

```diff
@@ -0,0 +1,44 @@
+package org.broadinstitute.orsp.api
+
+import com.google.gson.JsonParser
+import grails.converters.JSON
+import grails.rest.Resource
+import groovy.util.logging.Slf4j
+import org.broadinstitute.orsp.AuthenticatedController
+import org.broadinstitute.orsp.ConsentCollectionLink
+import org.broadinstitute.orsp.User
+import org.broadinstitute.orsp.utils.IssueUtils
+import org.springframework.web.multipart.MultipartFile
+
+@Slf4j
+@Resource(readOnly = false, formats = ['JSON', 'APPLICATION-MULTIPART'])
+class SampleConsentLinkController extends AuthenticatedController {
+
+    def pages() {
+        render(view: "/linkWizard/index")
+    }
+
+    def save() {
+        JsonParser parser = new JsonParser()
+        User user = getUser()
+        List<MultipartFile> files = request.multiFileMap.collect { it.value }.flatten()
+        ConsentCollectionLink consentCollectionLink = IssueUtils.getJson(ConsentCollectionLink.class, parser.parse(request.parameterMap["dataConsentCollection"].toString())[0])
```

Comment:
This looks a little brittle by assuming there is a 0-th element in the parsed parameter map. Some error handling around this would be safer.

Reply:
I think it's enough if I just move this line to the try-catch block, right?